pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1072,19 +1067,26 @@ int uprobe_lkm_seeker(struct pt_regs *ctx)
     }
 
     start_scan_time_init_shown_mods = bpf_ktime_get_ns();
-
-    if (init_shown_modules() != 0) {
-        tracee_log(ctx, BPF_LOG_LVL_ERROR, BPF_LOG_ID_UNSPEC, 1);
+    int ret = init_shown_modules();
+    if (ret != 0) {
+        tracee_log(ctx, BPF_LOG_LVL_ERROR, BPF_LOG_ID_UNSPEC, ret);
         return 1;
     }
 
+    // On first run, do a scan only relevant for modules that were inserted prior tracee started.
+    if (!hidden_old_mod_scan_done) {
```

Comment:
I am not sure how, but you probably need to make sure to do an atomic operation here.
I think `__sync_val_compare_and_swap` might work.

Reply:
Since this is a uprobe triggered based, and is triggered from userspace from a single location, this code can run only once at a time

Reply:
This is not a good reason not implementing it thread-safe, as there is a risk here.